src/dune_rules/coq_rules.ml

```diff
@@ -372,14 +397,17 @@ let parse_coqdep ~dir ~(boot_type : Bootstrap.t) ~coq_module
       Path.relative (Path.build (Coq_lib.src_root lib)) "Init/Prelude.vo"
       :: deps)
 
-let deps_of ~dir ~boot_type coq_module =
-  let stdout_to = Coq_module.dep_file ~obj_dir:dir coq_module in
-  Action_builder.dyn_paths_unit
-    (let open Action_builder.O in
-    let* boot_type = Resolve.Memo.read boot_type in
-    Action_builder.map
-      (Action_builder.lines_of (Path.build stdout_to))
-      ~f:(parse_coqdep ~dir ~boot_type ~coq_module))
+let deps_of ~dir ~boot_type ~plugin_deps coq_module =
+  let module_deps =
+    let stdout_to = Coq_module.dep_file ~obj_dir:dir coq_module in
+    Action_builder.dyn_paths_unit
+      (let open Action_builder.O in
+      let* boot_type = Resolve.Memo.read boot_type in
+      Action_builder.map
+        (Action_builder.lines_of (Path.build stdout_to))
+        ~f:(parse_coqdep ~dir ~boot_type ~coq_module))
+  in
+  Action_builder.O.(plugin_deps >>> module_deps)
```

Comment:
This is a spurious dep for all modules not doing `Declare ML Module` which are like > 99% of them in coq-universe.

Reply:
> What is the advantage of this approach? I can't see any. Morever, coqdep will keep emitting the .cmxs dep.

I plan to make sure that the cmxs dep actually comes from the declared plugins, this will allow us to catch attempts to load plugins not declared in the dune file at the dune level rather than Coq failing.